File: cpp/Shirt.cpp
#include "Shirt.hpp"

// Default constructor
Shirt::Shirt() : Clothing(), color(""), sleeveType("") {}

// Parameterized constructor
Shirt::Shirt(std::string size, std::string material, std::string gender, std::string color, std::string sleeveType)
    : Clothing(size, material, gender), color(color), sleeveType(sleeveType) {}

// Getters
std::string Shirt::getColor() {
    return color;
}

std::string Shirt::getSleeveType() {
    return sleeveType;
}

// Setters
void Shirt::setColor(std::string color) {
    this->color = color;
}

void Shirt::setSleeveType(std::string sleeveType) {
    this->sleeveType = sleeveType;
}

File: php/Shirt.php
<?php

require_once 'Clothing.php';

class Shirt extends Clothing {
    private $color;
    private $sleeveType;

    // Parameterized constructor
    public function __construct($size = "", $material = "", $gender = "", $color = "", $sleeveType = "") {
        parent::__construct($size, $material, $gender);
        $this->color = $color;
        $this->sleeveType = $sleeveType;
    }

    // Getters
    public function getColor() {
        return $this->color;
    }

    public function getSleeveType() {
        return $this->sleeveType;
    }

    // Setters
    public function setColor($color) {
        $this->color = $color;
    }

    public function setSleeveType($sleeveType) {
        $this->sleeveType = $sleeveType;
    }
}

?>
